modules/runner/src/integrationTest/java/org/apache/ignite/internal/Cluster.java

```diff
@@ -0,0 +1,346 @@
+        this.testInfo = testInfo;
+        this.workDir = workDir;
+        this.nodeBootstrapConfig = nodeBootstrapConfig;
+    }
+
+    /**
+     * Starts the cluster with the given number of nodes and initializes it.
+     *
+     * @param nodeCount Number of nodes in the cluster.
+     */
+    public void startAndInit(int nodeCount) {
+        if (started) {
+            throw new IllegalStateException("The cluster is already started");
+        }
+
+        List<CompletableFuture<IgniteImpl>> futures = IntStream.range(0, nodeCount)
+                .mapToObj(this::startClusterNode)
+                .collect(toList());
+
+        String metaStorageAndCmgNodeName = testNodeName(testInfo, 0);
+
+        IgnitionManager.init(metaStorageAndCmgNodeName, List.of(metaStorageAndCmgNodeName), "cluster");
+
+        for (CompletableFuture<IgniteImpl> future : futures) {
+            assertThat(future, willCompleteSuccessfully());
+
+            nodes.add(future.join());
+        }
+
+        started = true;
+    }
+
+    private CompletableFuture<IgniteImpl> startClusterNode(int nodeIndex) {
+        String nodeName = testNodeName(testInfo, nodeIndex);
+
+        String config = IgniteStringFormatter.format(nodeBootstrapConfig, BASE_PORT + nodeIndex, CONNECT_NODE_ADDR);
+
+        return IgnitionManager.start(nodeName, config, workDir.resolve(nodeName))
+                .thenApply(IgniteImpl.class::cast);
+    }
+
+    /**
+     * Returns an Ignite node (a member of the cluster) by its index.
+     */
+    public IgniteImpl node(int index) {
+        return nodes.get(index);
+    }
+
+    /**
+     * Returns a node that is not stopped and not knocked out (so it can be used to interact with the cluster).
+     */
+    public IgniteImpl entryNode() {
+        return IntStream.range(0, nodes.size())
+                .filter(index -> nodes.get(index) != null)
+                .filter(index -> !knockedOutIndices.contains(index))
+                .mapToObj(nodes::get)
+                .findAny()
+                .orElseThrow(() -> new IllegalStateException("There is no single alive node that would not be knocked out"));
+    }
+
+    /**
+     * Starts a new node with the given index.
+     *
+     * @param index Node index.
+     * @return Started node (if the cluster is already initialized, the node is returned when it joins the cluster; if it
+     *     is not initialized, the node is returned in a state in which it is ready to join the cluster).
+     */
+    public IgniteImpl startNode(int index) {
+        IgniteImpl newIgniteNode;
+
+        try {
+            newIgniteNode = startClusterNode(index).get(10, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+
+            throw new RuntimeException(e);
+        } catch (ExecutionException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+
+        nodes.set(index, newIgniteNode);
```

Comment:
If memory serves me right, then if the list is smaller than the index, then there will be an error.